src/wp-includes/link-template.php

```diff
@@ -1597,16 +1597,23 @@ function get_delete_post_link( $post = 0, $deprecated = '', $force_delete = fals
  * @since 2.3.0
  *
  * @param int|WP_Comment $comment_id Optional. Comment ID or WP_Comment object.
+ * @param string         $context    Optional. Could be 'display' or 'null'. Default null.
  * @return string|void The edit comment link URL for the given comment.
  */
-function get_edit_comment_link( $comment_id = 0 ) {
+function get_edit_comment_link( $comment_id = 0, $context = null ) {
 	$comment = get_comment( $comment_id );
 
 	if ( ! current_user_can( 'edit_comment', $comment->comment_ID ) ) {
 		return;
 	}
 
-	$location = admin_url( 'comment.php?action=editcomment&amp;c=' ) . $comment->comment_ID;
+	if ( 'display' === $context ) {
+		$action = 'comment.php?action=editcomment&c=';
+	} else {
+		$action = 'comment.php?action=editcomment&amp;c=';
+	}
```

Comment:
This needs to be the other way around: `display` is intended for including the URL in a link, i.e. "display" it. The other value would then be intended for usage of the URL outside of HTML markup.
```suggestion
	if ( 'display' === $context ) {
		$action = 'comment.php?action=editcomment&amp;c=';
	} else {
		$action = 'comment.php?action=editcomment&c=';
	}
```